crates/biome_css_analyze/src/lint/nursery/no_invalid_position_at_import_rule.rs

```diff
@@ -0,0 +1,99 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{AnyCssRule, CssImportAtRule, CssRuleList};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow invalid position @import rules.
+    ///
+    /// Any @import rules must precede all other valid at-rules and style rules in a stylesheet (ignoring @charset and @layer), or else the @import rule is invalid.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a {}
+    /// @import 'foo.css';
+    /// /** ↑
+    /// * This @import */
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// @import 'foo.css';
+    /// a {}
+    /// ```
+    ///
+    pub NoInvalidPositionAtImportRule {
+        version: "next",
+        name: "noInvalidPositionAtImportRule",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("no-invalid-position-at-import-rule")],
+    }
+}
+
+impl Rule for NoInvalidPositionAtImportRule {
+    type Query = Ast<CssRuleList>;
+    type State = CssImportAtRule;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+        let mut is_invalid_position = false;
+
+        for rule in node {
+            let any_css_at_rule = match rule {
+                AnyCssRule::CssAtRule(item) => item.rule().ok(),
+                _ => None,
+            };
+
+            if let Some(any_css_at_rule) = any_css_at_rule {
+                // Ignore @charset, @layer
+                if any_css_at_rule.as_css_charset_at_rule().is_some() {
+                    continue;
+                }
+                if any_css_at_rule.as_css_layer_at_rule().is_some() {
+                    continue;
+                }
+
+                let import_rule = any_css_at_rule.as_css_import_at_rule().cloned();
```

Comment:
Don't clone here. Clone when you actually need it, which is when you call `return`

Reply:
fix: https://github.com/biomejs/biome/pull/2717/commits/9ca1793d562753cd71d6f802f04552fc481b79ae